sonar-application/src/test/java/org/sonar/application/ClusterTest.java

```diff
@@ -46,13 +46,19 @@
   public void test_cluster() {
     Props props = initDefaultProps();
 
-    assertThat(
-      new Cluster(props).isActive()
-    ).isEqualTo(true);
+    Cluster cluster = new Cluster(props);
+    try {
+      assertThat(
+        cluster.isActive()
+      ).isEqualTo(true);
+    } finally {
+      cluster.hazelcastInstance.getLifecycleService().terminate();
```

Comment:
why isn't this `terminate` called anywhere in production code ?
does it mean hazelcast is not shutdown correctly when App is stopped?

Consider this:
Referring to my previous comment where class `Cluster` becomes `HazelcastConfigurer`.
I would do the following:

1. name the class `HazelcastWrapper` rather than `HazelcastConfigurer`
2. rename method `configure` to `configureAndStart` (since it does both)
3. make class `HazelcastWrapper` an `AutoCloseable` and method `close` would call `getLifecycleService().terminate()`
4. update App to end up with the following code:
```
ClusterProperties clusterProperties = new ClusterProperties(props);
clusterProperties.validate();
try (HazelcastWrapper hazelcastWrapper = new HazelcastWrapper(clusterProperties)) {
    hazelcastWrapper.configure();
    
    [...]
    app.start(props);
}
```

I'm pretty sure this way we won't have to terminate Hazelcast in tests explicitly (and that in production, it will now be shutdown appropriately)